<?xml version="1.0"?>
<robot name="scan_model"
  xmlns:xacro="http://www.ros.org/wiki/xacro">

  <!-- 매크로 정의 -->
  <xacro:macro name="scan_model" params="file_name">
    <link name="scan_model">
      <visual>
        <geometry>
          <mesh filename="package://nrs_ver2/mesh/${file_name}.stl" />
        </geometry>
        <material name="blue">
          <color rgba="0.0 0.0 0.5 1.0" />
        </material>
      </visual>
    </link>
    <!-- 원래 위치에서의 링크 정의 -->
    <link name="cad_model">
      <visual>
        <geometry>
          <mesh filename="package://nrs_ver2/mesh/$(arg file_name).stl" />
        </geometry>
        <material name="transparent">
          <color rgba="0.5 0.5 0.5 0.0" />
        </material>
      </visual>
    </link>


    <!-- base_link와 scan_model 연결 -->
    <joint name="base_to_cad_model" type="fixed">
      <parent link="base_link"/>
      <child link="cad_model"/>
      <origin xyz="0 0 0" rpy="0 0 0" />
    </joint>


    <!-- base_link와 scan_model 연결 -->
    <joint name="cad_to_scan_model" type="fixed">
      <parent link="cad_model"/>
      <child link="scan_model"/>
    <origin xyz="0.7612783054351273 -0.03097931096065283 0.12363673620305149" rpy="0.024916701375427223 0.013877853294511456 2.3236770153618296" />
    </joint>
  </xacro:macro>
</robot>